starlite/asgi/routing_trie/traversal.py

```diff
@@ -30,37 +51,43 @@ def traverse_route_map(
     path_params: List[Tuple[str, Any]] = []
     path_components = [p for p in path.split("/") if p]
 
-    for i, component in enumerate(path_components):
+    for component in path_components:
         if component in current_node.child_keys:
             current_node = current_node.children[component]
             continue
 
-        if current_node.path_param_definition:
-            param_definition = current_node.path_param_definition
-            if param_definition.type is Path:
-                path_params.append((param_definition.name, normalize_path("/".join(path_components[i:]))))
-                return current_node, path_params, path
-
-            path_params.append((param_definition.name, param_definition.parser(component)))
-            current_node = current_node.children[PathParameterSentinel]
-
-        continue
+        if current_node.path_type_path_param_definition:
+            path_params.append(
+                (
+                    current_node.path_type_path_param_definition.name,
+                    normalize_path(removeprefix(path, current_node.path)),
```

Comment:
Another use for `RouteTrieNode.path` - this is slightly more efficient with `removeprefix()`:
```
$ poetry run python -m timeit -s 'from starlite.utils import normalize_path' -s 'path="/a/b/c/d/e"' -s 'pre="/a/b"' 'path.removeprefix(pre)'
10000000 loops, best of 5: 29.2 nsec per loop
$ poetry run python -m timeit -s 'from starlite.utils import normalize_path' -s 'components=["c","d","e"]' '"/".join(components)'
5000000 loops, best of 5: 42.9 nsec per loop
```
Although, `normalize_path()` dwarfs the difference in the two.